### PR TITLE
Allow for installing bundles from the registry

### PR DESCRIPTION
**Pull request by vanstee**
This allows us to install things from the registry like:

`cogctl bundle install heroku` - which will install the latest version of the heroku bundle
`cogctl bundle install heroku:0.1.0` - which will install the 0.1.0 version of the heroku bundle
`cogctl bundle install bundles/heroku/config.yaml` - which will install the heroku bundle from the config provided

NOTE: I may need to change the hardcoded url once we have that hosted and running correctly.

### CODE REVIEW

File: lib/cogctl/actions/bundle/install.ex [hunk 1]
```diff
@@ -28,15 +29,15 @@ defmodule Cogctl.Actions.Bundle.Install do
   """
 
   def option_spec do
-    [{:file, :undefined, :undefined, :string, 'Path to your bundle config file (required)'},
+    [{:bundle_or_path, :undefined, :undefined, :string, 'Name and verison of registered bundle or local path to bundle config (required). Examples: heroku, heroku:0.4.0, bundles/heroku/config.yaml'},
```
Comment (christophermaier) [inline lib/cogctl/actions/bundle/install.ex]: I though we were going with the version being a standalone optional argument?

File: lib/cogctl/actions/bundle/install.ex [hunk 2]
```diff
@@ -3,6 +3,7 @@ defmodule Cogctl.Actions.Bundle.Install do
 
   alias CogApi.HTTP.Client
   alias Cogctl.Table
+  alias Cogctl.BundleRegistry
   import Cogctl.Actions.Bundle.Helpers, only: [field: 2]
 
   @template_extension ".mustache"
```
Comment (christophermaier) [inline lib/cogctl/actions/bundle/install.ex]: This part is going to need some revisiting, as well as the `moduledoc` below.

File: lib/cogctl/actions/bundle/install.ex [hunk 3]
```diff
@@ -80,18 +81,28 @@ defmodule Cogctl.Actions.Bundle.Install do
...
-  defp parse_string_or_file(file) do
-    if File.exists?(file) do
-      Spanner.Config.Parser.read_from_file(file)
-    else
-      Spanner.Config.Parser.read_from_string(file)
+  defp parse_bundle_or_path(bundle_or_path) do
+    cond do
+      File.exists?(bundle_or_path) ->
+        Spanner.Config.Parser.read_from_file(bundle_or_path)
+      match?({:ok, _}, Poison.decode(bundle_or_path)) ->
```
Comment (christophermaier) [inline lib/cogctl/actions/bundle/install.ex]: I think I missed how standard input gets piped down here... Also, that usecase doesn't seem to be described in the doc strings above.

File: lib/cogctl/bundle_registry.ex [hunk 4]
```diff
@@ -0,0 +1,15 @@
+defmodule Cogctl.BundleRegistry do
```
Comment (christophermaier) [inline lib/cogctl/bundle_registry.ex]: Maybe we should have an API in Cog that we hit to install the bundle instead, since this is duplicated from Cog.

**Comment by vanstee** [comment]
This is going to fail until we bump the `cog_api` dep once this is merged: https://github.com/operable/cog-api-client/pull/142

**Pull request merged**